js/packages/teams-ai/src/Application.ts

```diff
@@ -1047,6 +1132,23 @@ function createMessageReactionSelector(event: MessageReactionEvents): RouteSelec
     }
 }
 
+/**
+ * Returns a selector function that indicates whether the bot should initiate a sign in.
+ * @param {boolean | Selector} startSignIn A boolean or function that indicates whether the bot should initiate a sign in.
+ * @returns {Selector} A selector function that returns true if the bot should initiate a sign in.
+ */
+function createSignInSelector(startSignIn?: boolean | Selector): Selector {
```

Comment:
Why is boolean needed? shouldn't it just be the selector?

Reply:
`boolean` is so that users don't have to implement a selector function just to enable/disable it for all incoming activities.

Reply:
This seems clunky to me, but I don't have simple suggested alternative. Maybe we can bring this up for discussion briefly in a meeting? Doesn't this mean that if they put `false` it will still initiate `startSignIn`? So then isn't it superfluous?

@aacebo do you have thoughts / think this is good as-is?

Reply:
If they put `false` then a selector function will be returned the resolves to false. This means that auth will not be triggered.

Also, this pattern of overloading the parameters is used in virtually every selector creation method:

![image](https://github.com/microsoft/teams-ai/assets/115390646/106af29c-4dc4-4399-b13a-f1d39c8a11eb)

Reply:
Resolving - confusion was that I was magically skipping over line 1144 when trying to parse. Thanks for explaining.